pkgs/by-name/ta/tabby-terminal/package.nix

```diff
@@ -0,0 +1,258 @@
+      ''
+        runHook preBuild
+
+        mkdir -p http-cache/v${electronPackage.version}
+        cp $electronHeaders http-cache/v${electronPackage.version}/node-v${electronPackage.version}-headers.tar.gz
+        cp $electronHeadersSHA http-cache/v${electronPackage.version}/SHASUMS256.txt
+        http-server http-cache &
+      ''
+      # Rebuild all the needed packages using electron-rebuild
+      + builtins.concatStringsSep "\n" (
+        map (pkg: ''
+          yarn --offline electron-rebuild -v ${electronPackage.version} -m ${pkg} -f -d "http://localhost:8080"
+        '') rebuildPkgs
+      )
+      + ''
+        kill $!
+        yarn build
+
+        runHook postBuild
+      '';
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out/share/tabby
+      mkdir -p $out/bin
+      cp -r . $out/share/tabby
+
+      # Tabby needs TABBY_DEV to run manually with electron
+      makeWrapper "${electronPackage}/bin/electron" "$out/bin/tabby" --add-flags "$out/share/tabby/app" --set TABBY_DEV 1
+
+      for iconsize in 16 32 64 128
+      do
+        size=$iconsize"x"$iconsize
+        mkdir -p $out/share/icons/hicolor/$size/apps
+        ln -s $out/share/tabby/build/icons/"$size".png $out/share/icons/hicolor/$size/apps/tabby.png
+      done
+
+      runHook postInstall
+    '';
+  };
+
+  builtinPlugins = [
+    "tabby-core"
+    "tabby-settings"
+    "tabby-terminal"
+    "tabby-web"
+    "tabby-community-color-schemes"
+    "tabby-ssh"
+    "tabby-serial"
+    "tabby-telnet"
+    "tabby-local"
+    "tabby-electron"
+    "tabby-plugin-manager"
+    "tabby-linkifier"
+  ];
+
+  packages = builtinPlugins ++ [
+    "."
+    "app"
+    "web"
+    "tabby-web-demo"
+  ];
+
+  pkgHashes = {
+    "." = "sha256-W4T9Vaow37yBD08UFv7Wby2OKiz/Lp70OXddGjTckxU=";
+    app = "sha256-pTHCyxx7pXKyITKzV3nrtdPX7ZUc+7oBMO3E8Py4Uiw=";
+    web = "sha256-kdER/yB8O7gfWnLZ/rNl4ha1eNnypcVmS1L7RrFCn0Q=";
+    tabby-core = "sha256-Z42TZeE0z96ZRtIFIgGbQyv8gtGY/Llya/Dhs8JtuWo=";
+    tabby-local = "sha256-GmVoeKxF8Sj55fDPN4GhwGVXoktdiwF3EFYTJHGO/NQ=";
+    tabby-settings = "sha256-7t9mXsMqU892fLHyHmivgDxECSVn8KgRNAz9E2nKC/I=";
+    tabby-electron = "sha256-JDEsn+7xOcoKFyOCFeClFqcD2NJWwRA6uyM8SwxKW8c=";
+    tabby-web-demo = "sha256-i8UoUMb5m+rf/73eKPm2S0v6cs8qSqy6lRjnZ6GoO/k=";
+    tabby-linkifier = "sha256-78wgw6BbN/GtRMYeJF9svn9hn82bTZj4+8TXf/rAC64=";
+    tabby-web = "sha256-ErhWM0jiVK4PBosBz4IHi1xiemAzRuk/EE8ntyhO2PE=";
+    tabby-telnet = "sha256-J8nBBUxwTdigcdohEF6dw8+EHRBUm8O1SLM9oDB3VaA=";
+    tabby-ssh = "sha256-8PRf0F6Q3Hoqxiz6lcoFZn0z9EAyeTNNuSh6gli35+U=";
+    tabby-community-color-schemes = "sha256-oZgyP0hTU9bxszOVg3Bmiu6yos2d2Inc1Do8To4z8GQ=";
+    tabby-plugin-manager = "sha256-CrgsIGg834A+WQh7o07Quv+SSFqYxPMugZsSOHCrdPU=";
+    tabby-serial = "sha256-sg/CJnlkUcohFgmY6xGE79WG5mmx9jh196mb8iVCk6g=";
+    tabby-terminal = "sha256-LS30V7iqLI0sEm3xlnMFtMnIxhALOMjQ148+zR0NyqU=";
+  };
```

Comment:
I wonder if this should be in a separate JSON file so that we can add an update script (either now or in the future) that would calculate these hashes for us